gax-java/gax/src/main/java/com/google/api/gax/rpc/StubSettings.java

```diff
@@ -289,6 +300,7 @@ protected Builder(ClientContext clientContext) {
         this.streamWatchdogCheckInterval = clientContext.getStreamWatchdogCheckInterval();
         this.tracerFactory = clientContext.getTracerFactory();
         this.quotaProjectId = getQuotaProjectIdFromClientContext(clientContext);
+        this.gdchApiAudience = clientContext.getGdchApiAudience();
```

Comment:
This is probably a larger design issue, I'm not sure why we need to create `StubSettings` from `ClientContext`. Technically we only need to set `gdchApiAudience` to `StubSettings`, use it to get credentials when create `ClientContext`. If we don't need to support this case, we wouldn't need to expose getter and setters for `gdchApiAudience` in `ClientContext`.

Reply:
I don't have context on why, but seems like creation of the StubSettings from ClientContext is an existing historical thing and therefore it probably make sense to carry forward?

Reply:
Yes, Diego and I discussed offline and we agreed to keep it.